packages/astro/src/core/build/index.ts

```diff
@@ -214,9 +209,15 @@ class AstroBuilder {
 		const { internals, ssrOutputChunkNames, contentFileNames } = await viteBuild(opts);
 
 		const hasServerIslands = this.settings.serverIslandNameMap.size > 0;
-		// Error if there are server islands but no adapter provided.
-		if (hasServerIslands && this.settings.buildOutput !== 'server') {
-			throw new AstroError(AstroErrorData.NoAdapterInstalledServerIslands);
+		if (hasServerIslands) {
+			if (this.settings.buildOutput === 'server') {
+				// Can't be injected in runHookRoutesResolved because serverIslandNameMap is updated
+				// in a vite plugin transform
+				injectServerIslandRoute(this.settings.config, this.manifest);
```

Comment:
Looks like the data is not passed to the manifest, probably a matter of changing when this is called